### PR TITLE
Fix cache-only for the caches

### PR DESCRIPTION
**Pull request by JoviDeCroock**
## Summary

`cache-only` currently dispatches a network-call when the result isn't in cache.

## Set of changes

- Add a `cache-only` filter so the operation does not get forwarded in doc-cache.
- Moves the `graphcache` operations outside of `forward` so we can add the same filter as introduced in doc-cache

Issue: https://github.com/FormidableLabs/next-urql/issues/39#issuecomment-588901815

### CODE REVIEW

File: exchanges/graphcache/src/cacheExchange.ts [hunk 1]
```diff
@@ -342,15 +342,15 @@ export const cacheExchange = (opts?: CacheExchangeOpts): Exchange => ({
...
-        ])
-      ),
+      merge([
+        pipe(
+          inputOps$,
+          filter(op => !isCacheableQuery(op))
+        ),
+        cacheOps$,
+      ]),
+      filter(op => !(op.operationName === 'query' && op.context.requestPolicy === 'cache-only')),
```
Comment (andyrichardson) [inline exchanges/graphcache/src/cacheExchange.ts]: Is it fair to assume we'll never get a non-query operation with a cachePolicy?

```suggestion
      filter(op => op.context.requestPolicy !== 'cache-only'),
```

File: packages/core/src/exchanges/cache.ts [hunk 2]
```diff
@@ -86,6 +86,7 @@ export const cacheExchange: Exchange = ({ forward, client }) => {
         ),
       ]),
       map(op => addMetadata(op, { cacheOutcome: 'miss' })),
+      filter(op => !(op.operationName === 'query' && op.context.requestPolicy === 'cache-only')),
```
Comment (andyrichardson) [inline packages/core/src/exchanges/cache.ts]: Woah, this was a bug in the default cache too? 😱

Comment (kitten) [reply packages/core/src/exchanges/cache.ts]:
```suggestion
      filter(op => op.operationName !== 'query' || op.context.requestPolicy !== 'cache-only'),
```

File: .changeset/fluffy-guests-begin.md [hunk 3]
```diff
@@ -0,0 +1,6 @@
+---
+'@urql/exchange-graphcache': patch
+'@urql/core': patch
+---
+
+Prevents `cache-only` operations from getting forwarded outside of the cache when no result was found
```
Comment (kitten) [inline .changeset/fluffy-guests-begin.md]: Let's make sure this is user-friendly without implementation details?

```suggestion
Fix `cache-only` operations being forwarded and triggering fetch requests
```

**Comment by subpopular** [comment]
I am still seeing a network request with `{requestPolicy 'cache-only'}` on my query on latest version.

**Comment by kitten** [comment]
@subpopular I'm not quite sure what you're looking for but it sounds like maybe you'll want to open a new issue with a question? :) https://github.com/FormidableLabs/urql/issues/new?labels=question+%F0%9F%99%8B&template=Question.md&title=

**Pull request merged**